.travis.yml

```diff
@@ -3,42 +3,49 @@ services: docker
 language: bash
 addons:
     apt:
-        config:
-            retries: true
         update: true
         packages:
-            - jq
             - rpm2cpio
             - cpio
 env:
-    global:
-        - VERSION=4.0.0-5
-        # See qemu-user-static's RPM spec file on Fedora to check the new version.
-        # https://src.fedoraproject.org/rpms/qemu/blob/master/f/qemu.spec
-        - DOCKER_SERVER=docker.io
-        # - DOCKER_SERVER=quay.io
-        # Git repository
-        - REPO=multiarch/qemu-user-static
-        # Container repository
-        - DOCKER_REPO=$DOCKER_SERVER/multiarch/qemu-user-static
-        # - DOCKER_REPO=$DOCKER_SERVER/your_username/qemu-user-static
```

Comment:
Please keep `VERSION` (For adhoc pipline), `DOCKER_SERVER`, `REPO`, `DOCKER_REPO`.
In general, it's not only for this project, I think that a good strategy to send pull-request, is "1 pull-request, 1 feature".
You changed a lot including some features.

The reason why `1 pull-request, 1 feature` strategy is better is you can proceed your work understanding project people's needs and intentions towards your final goal.
The output becomes "1 commit 1 feature". The merit of this is if the new commit has a issue, we can revert the commit later.

Though I understand the each feature in this PR is great.

I am thinking how we can proceed this PR.

Reply:
Everything was commented out, and defaults were used.  I thought that in a case like this it's simpler to start with a cleaner slate, and then gradually add features, as they become needed at the time.

Also, not sure what "adhoc pipline" means, and how and what version to add 🤔